app/src/main/java/org/mozilla/focus/widget/TelemetrySwitchPreference.java

```diff
@@ -41,13 +39,20 @@ public TelemetrySwitchPreference(Context context, AttributeSet attrs, int defSty
     }
 
     private void init() {
-        setWidgetLayoutResource(R.layout.preference_telemetry);
+//        setWidgetLayoutResource(R.layout.preference_telemetry);
 
         // We are keeping track of the preference value ourselves.
         setPersistent(false);
     }
 
     @Override
+    protected View onCreateView(ViewGroup parent) {
+        super.onCreateView(parent);
+        LayoutInflater li = (LayoutInflater)getContext().getSystemService( Context.LAYOUT_INFLATER_SERVICE );
+        return li.inflate( R.layout.preference_telemetry, parent, false);
```

Comment:
What's the difference if we replace the layout reference in settings.xml from focus_preference to this preference_telemetry? This override could be removed completely, correct?

Reply:
I think replace the layout reference in settings.xml from focus_preference is just replace a part of base preference layout. I will try which one is correct.